src/main/java/com/force/api/ExtendedApiVersion.java

```diff
@@ -0,0 +1,95 @@
+package com.force.api;
+
+
+/**
+ * represents a single supported API version as reported by the <code>/services/data</code> resource
+ * for the instance endpoint returned on authentication.
+ *
+ * {@link com.frejo.api.ApiVersion} was originally used to set the current API version and manage the complete list
+ * of versions as an enum. It will likely be phased out in favor of using plain version strings along with this class
+ * for more extended version information.
+ *
+ * This is a tweak to an original proposal by @cswendrowski.
+ *
+ */
+public class ExtendedApiVersion implements Comparable<ExtendedApiVersion> {
+
+    public enum Season {
+        WINTER(),
+        SPRING(),
+        SUMMER();
+    }
+
+    public ExtendedApiVersion() {}
+
+    public ExtendedApiVersion(float version) {
+        this.version = version;
+    }
+
+    public ExtendedApiVersion(Season s, int y) {
+        // Thank you @cswendrowski for this neat algorithm
+        this.version = (float) (3*(y - 2004) + s.ordinal() - 1);
+        if(version < 20) {
+            throw new IllegalArgumentException("Only Winter 2011 (v20.0) and later versions are supported");
```

Comment:
I originally threw an exception for versions < 1 to prevent the need for any future maintenance of this check. I see the draw of setting it to the current Oldest Version supported, but the User can fetch that value via the SupportedVersions list. This is technically a better value for the floor since the versions should never go backward, but it might be a misleading error message.